components/automate-cli/pkg/verifyserver/services/batchcheckservice/batchcheckservice.go

```diff
@@ -32,13 +50,230 @@ func (ss *BatchCheckService) BatchCheck(checks []string, config models.Config) m
+	}
 	if len(remoteChecks) > 0 {
 		for key, value := range getRemoteCheckResp(ss, remoteChecks, config) {
 			checkTriggerRespMap[key] = value
 		}
 	}
-	return constructBatchCheckResponse(checkTriggerRespMap, append(bastionChecks, remoteChecks...))
+
+	if len(successfullyStartedMockServers) > 0 {
+		for _, successfullyStartedMockServer := range successfullyStartedMockServers {
+			go ss.StopMockServerOnHostAndPort(successfullyStartedMockServer.Host, successfullyStartedMockServer.Protocol, successfullyStartedMockServer.Port)
+		}
+	}
+
+	return constructBatchCheckResponse(checkTriggerRespMap, append(bastionChecks, remoteChecks...)), nil
+}
+
+func (ss *BatchCheckService) startMockServerIfNeeded(remoteChecks []string, config models.Config) ([]models.StartMockServerFromBatchServiceResponse, []models.StartMockServerFromBatchServiceResponse, error) {
+	successfullyStartedMockServers := []models.StartMockServerFromBatchServiceResponse{}
+	notStartedMockServers := []models.StartMockServerFromBatchServiceResponse{}
+
+	if len(remoteChecks) > 0 {
+		startMockServer, err := ss.shouldStartMockServer(remoteChecks, config)
+		if err != nil {
+			return successfullyStartedMockServers, notStartedMockServers, err
+		}
+		if startMockServer {
+			successfullyStartedMockServers, notStartedMockServers = ss.StartMockServer(remoteChecks, config.Hardware)
+		}
+	}
+	return successfullyStartedMockServers, notStartedMockServers, nil
+}
+
+func (ss *BatchCheckService) shouldStartMockServer(remoteChecks []string, config models.Config) (bool, error) {
+	if stringutils.SliceContains(remoteChecks, constants.FIREWALL) || stringutils.SliceContains(remoteChecks, constants.FQDN) {
+		deploymentState, err := ss.getDeploymentState(config)
+		if err != nil {
+			ss.log.Error("Error while calling status api from batch check service:", err)
+			return false, err
+		}
+		return deploymentState == constants.PRE_DEPLOY, nil
+	}
+	return false, nil
+}
+
+func (ss *BatchCheckService) StartMockServer(remoteChecks []string, hardwareDetails models.Hardware) ([]models.StartMockServerFromBatchServiceResponse, []models.StartMockServerFromBatchServiceResponse) {
+	nodeTypePortMap := map[string]map[string][]int{
+		constants.AUTOMATE: {
+			constants.TCP:   []int{},
+			constants.HTTP:  []int{},
+			constants.UDP:   []int{},
+			constants.HTTPS: []int{},
+		},
+		constants.CHEF_INFRA_SERVER: {
+			constants.TCP:   []int{},
+			constants.HTTP:  []int{},
+			constants.UDP:   []int{},
+			constants.HTTPS: []int{},
+		},
+		constants.POSTGRESQL: {
+			constants.TCP:   []int{},
+			constants.HTTP:  []int{},
+			constants.UDP:   []int{},
+			constants.HTTPS: []int{},
+		},
+		constants.OPENSEARCH: {
+			constants.TCP:   []int{},
+			constants.HTTP:  []int{},
+			constants.UDP:   []int{},
+			constants.HTTPS: []int{},
+		},
+	}
+	for _, check := range remoteChecks {
+		resp := ss.GetPortsToOpenForCheck(check)
+		if len(resp) != 0 {
+			constructUniquePortMap(resp, &nodeTypePortMap)
+		}
+	}
```

Comment:
can we create a function like `createNodeTypeMap`

Reply:
Done